### PR TITLE
feat: support generating circom witness

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR enables generating circom witness.

Please test this by running this command!

```shell
cd vendors/circom
bazel run //examples:compute_witness_multiplier_3
```

You can find `multiplier_3.wtns` in `bazel-bin/examples/compute_witness_multiplier_3.runfiles/kroma_network_circom`.

See also https://docs.circom.io/getting-started/computing-the-witness/.

### CODE REVIEW

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by ashjeong — COMMENTED** [review]
https://github.com/kroma-network/tachyon/pull/342/commits/1a01978d39df2eb4a2e8191f36e5c48c04e2e903
Maybe don't include these files copied from [iden3/circom/code_producers/src/c_elements](https://github.com/iden3/circom/tree/v2.1.8/code_producers/src/c_elements) in our source code, but just rely on them as a dependency or such, since we cannot claim full ownership as we didn't really modify much of anything.

**Comment by chokobole** [comment]
> [1a01978](https://github.com/kroma-network/tachyon/commit/1a01978d39df2eb4a2e8191f36e5c48c04e2e903) Maybe don't include these files copied from [iden3/circom/code_producers/src/c_elements](https://github.com/iden3/circom/tree/v2.1.8/code_producers/src/c_elements) in our source code, but just rely on them as a dependency or such, since we cannot claim full ownership as we didn't really modify much of anything.

I'll handle this please don't merge this until I am done with this!

**Comment by chokobole** [comment]
@ashjeong I modified to generate `fr.cpp`, `fr.hpp` and `fr.asm` from a generator and add some diff files to track the changes for better maintenance.

**Review by dongchangYoo — APPROVED** [review]
LGTM

File: vendors/circom/circomlib/generated/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+This is taken and modified from [iden3/circom/code_producers/src/c_elements](https://github.com/iden3/circom/tree/v2.1.8/code_producers/src/c_elements).
+
+## Modifications
+
+- Modified to remove errors from compilation.
+- `loadCircuit` and `writeBinWitness` are moved to [common/calcwit.hpp](/vendors/circom//circomlib/generated/common/calcwit.hpp) and [common/calcwit.hpp](/vendors/circom//circomlib/generated/common/calcwit.cpp).
+- `loadWitness` is created instead of `loadJson`, which loads the witness from the `absl::flat_hash_map<>`.
+
+See the followings for more details.
```
Comment (ashjeong) [inline vendors/circom/circomlib/generated/README.md:11]:
``` suggestion
See the following files for more details on the modifications.
```
["Following" is an adjective](https://www.quora.com/What-should-I-say-Which-of-the-following-or-Which-of-the-followings) and, therefore, never plural,

**Review by ashjeong — APPROVED** [review]
LGTM

**Pull request merged**